sudo: python: command not found
sudo: python: command not found
sudo: python: command not found
sudo: python: command not found
sudo: python: command not found
sudo: python: command not found
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain/document_loaders/__init__.py:36: LangChainDeprecationWarning: Importing document loaders from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.document_loaders import PyPDFLoader`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain/vectorstores/__init__.py:35: LangChainDeprecationWarning: Importing vector stores from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.vectorstores import Chroma`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain/chat_models/__init__.py:31: LangChainDeprecationWarning: Importing chat models from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.chat_models import ChatOpenAI`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:117: LangChainDeprecationWarning: The class `langchain_community.chat_models.openai.ChatOpenAI` was deprecated in langchain-community 0.0.10 and will be removed in 0.2.0. An updated version of the class exists in the langchain-openai package and should be used instead. To use it run `pip install -U langchain-openai` and import as `from langchain_openai import ChatOpenAI`.
  warn_deprecated(
INFO:waitress:Serving on http://0.0.0.0:8080
sudo: waitress-serve: command not found
WARNING:waitress:total open connections reached the connection limit, no longer accepting new connections
INFO:waitress:total open connections dropped below the connection limit, listening again
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:117: LangChainDeprecationWarning: The function `__call__` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use invoke instead.
  warn_deprecated(
sudo: waitress-serve: command not found
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain/document_loaders/__init__.py:36: LangChainDeprecationWarning: Importing document loaders from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.document_loaders import PyPDFLoader`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain/vectorstores/__init__.py:35: LangChainDeprecationWarning: Importing vector stores from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.vectorstores import Chroma`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain/chat_models/__init__.py:31: LangChainDeprecationWarning: Importing chat models from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.chat_models import ChatOpenAI`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:117: LangChainDeprecationWarning: The class `langchain_community.chat_models.openai.ChatOpenAI` was deprecated in langchain-community 0.0.10 and will be removed in 0.2.0. An updated version of the class exists in the langchain-openai package and should be used instead. To use it run `pip install -U langchain-openai` and import as `from langchain_openai import ChatOpenAI`.
  warn_deprecated(
INFO:waitress:Serving on http://0.0.0.0:8080
/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:117: LangChainDeprecationWarning: The function `__call__` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use invoke instead.
  warn_deprecated(
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
ERROR:app:Exception on /upload [POST]
Traceback (most recent call last):
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/flask/app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/flask/app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/flask/app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/flask/app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/root/bidBoost-RAG-api/app.py", line 116, in upload_files
    conversation_chain = initialize_app()
  File "/root/bidBoost-RAG-api/app.py", line 63, in initialize_app
    vectordb = Chroma.from_documents(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_community/vectorstores/chroma.py", line 778, in from_documents
    return cls.from_texts(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_community/vectorstores/chroma.py", line 736, in from_texts
    chroma_collection.add_texts(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_community/vectorstores/chroma.py", line 275, in add_texts
    embeddings = self._embedding_function.embed_documents(texts)
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_openai/embeddings/base.py", line 517, in embed_documents
    return self._get_len_safe_embeddings(texts, engine=engine)
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/langchain_openai/embeddings/base.py", line 333, in _get_len_safe_embeddings
    response = self.client.create(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/resources/embeddings.py", line 113, in create
    return self._post(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/_base_client.py", line 1208, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/_base_client.py", line 897, in request
    return self._request(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/_base_client.py", line 973, in _request
    return self._retry_request(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/_base_client.py", line 1021, in _retry_request
    return self._request(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/_base_client.py", line 973, in _request
    return self._retry_request(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/_base_client.py", line 1021, in _retry_request
    return self._request(
  File "/root/bidBoost-RAG-api/.venv/lib/python3.10/site-packages/openai/_base_client.py", line 988, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.RateLimitError: Error code: 429 - {'error': {'message': 'Rate limit reached for text-embedding-ada-002 in organization org-Vg3rRAOOS6BrcZ8iMHRPVrc8 on tokens per min (TPM): Limit 1000000, Used 794265, Requested 272795. Please try again in 4.023s. Visit https://platform.openai.com/account/rate-limits to learn more.', 'type': 'tokens', 'param': None, 'code': 'rate_limit_exceeded'}}
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
WARNING:pypdf.generic._data_structures:Multiple definitions in dictionary at byte 0x4cd28 for key /PageMode
sudo: waitress-serve: command not found
sudo: waitress-serve: command not found
sudo: waitress-serve: command not found
sudo: python: command not found
sudo: waitress-serve: command not found
